test/check-application

```diff
@@ -2065,6 +2069,213 @@ class TestFiles(testlib.MachineCase):
+        b.click("#dropdown-menu")
+        b.click("#copy-item")
+        b.go("files#/?path=/home/foouser")
+        self.assert_last_breadcrumb("foouser")
+        b.click("#dropdown-menu")
+        b.click("#paste-item")
+        b.wait_visible("#paste-owner-modal")
+        b.select_from_dropdown("#paste-owner-select", "root")
+        b.click("#paste-owner-modal button.pf-m-primary")
+        b.wait_in_text("[data-item='kbdCopy'] .item-owner", "root")
+        b.go("files#/?path=/home/foouser/kbdCopy")
+        self.assert_last_breadcrumb("kbdCopy")
+        b.wait_in_text("[data-item='loaded'] .item-owner", "root")
+        b.wait_in_text("[data-item='newfile'] .item-owner", "root")
+
+        # Copy file with different user, group
+        b.go("files#/?path=/home/admin")
+        self.assert_last_breadcrumb("admin")
+        m.execute("""
+            echo 'adminfoo' > /home/admin/adminfoofile
+            chown admin:foouser /home/admin/adminfoofile
+        """)
+        b.click("[data-item='adminfoofile']")
+        b.click("#dropdown-menu")
+        b.click("#copy-item")
+        b.go("files#/?path=/home/foouser")
+        self.assert_last_breadcrumb("foouser")
+        b.click("#dropdown-menu")
+        b.click("#paste-item")
+        b.wait_visible("#paste-owner-modal")
+        b.select_from_dropdown("#paste-owner-select", "admin:foouser (original owner)")
+        b.click("#paste-owner-modal button.pf-m-primary")
+        b.wait_in_text("[data-item='adminfoofile'] .item-owner", "admin:foouser")
+        self.assertEqual(m.execute("head -n 1 /home/foouser/adminfoofile"), "adminfoo\n")
+
+        b.go("files#/?path=/home/admin")
+        self.assert_last_breadcrumb("admin")
+        b.click("[data-item='newdir']")
+        b.click("#dropdown-menu")
+        b.click("#copy-item")
+        b.go("files#/?path=/home/foouser")
+        self.assert_last_breadcrumb("foouser")
+        b.click("#dropdown-menu")
+        b.click("#paste-item")
+        b.wait_visible("#paste-owner-modal")
+        b.click("button.pf-m-link")
+        b.wait_not_present("#paste-owner-modal")
+        b.wait_not_present("[data-item='newdir']")
+
+        # Test where copy fails
+        # Add +i attribute on the directory which makes it immutable
+        m.execute("""
+            runuser -u foouser mkdir /home/foouser/copyfail
+            chattr +i /home/foouser/copyfail
+        """)
+        self.addCleanup(m.execute, "chattr -i /home/foouser/copyfail")
+        b.go("files#/?path=/home/admin")
+        self.assert_last_breadcrumb("admin")
+        b.click("[data-item='newfile']")
+        b.click("#dropdown-menu")
+        b.click("#copy-item")
+        b.go("files#/?path=/home/foouser/copyfail")
+        self.assert_last_breadcrumb("copyfail")
+        b.click("#dropdown-menu")
+        b.click("#paste-item")
+        b.click("#paste-owner-modal button.pf-m-primary")
+        b.wait_in_text("h4.pf-v5-c-alert__title", "Pasting failed")
```

Comment:
Can you please check that this has a sensible error message, i.e. that the target dir and something like "permission denied" are in it?